esdb/containers_test.go

```diff
@@ -109,122 +108,100 @@ func IsESDBVersion20() bool {
 	})
 }
 
-func getDockerOptions() *dockertest.RunOptions {
+func getContainerRequest() (*EventStoreDockerConfig, *testcontainers.ContainerRequest, error) {
 	config := readEnvironmentVariables(defaultEventStoreDockerConfig)
 
-	var envs []string
-	if isInsecure, exists := os.LookupEnv("EVENTSTORE_INSECURE"); exists {
-		envs = append(envs, fmt.Sprintf("EVENTSTORE_INSECURE=%s", isInsecure))
-	}
-
-	if len(envs) == 0 {
-		envs = append(envs, fmt.Sprintf("EVENTSTORE_INSECURE=%s", "true"))
-	}
-
-	return &dockertest.RunOptions{
-		Repository:   config.Repository,
-		Tag:          config.Tag,
-		ExposedPorts: []string{config.Port},
-		Env:          envs,
-	}
-}
-
-func (container *Container) Close() {
-	err := container.Resource.Close()
-	if err != nil {
-		panic(err)
-	}
-}
-
-func getDatabase(t *testing.T, options *dockertest.RunOptions) *Container {
-	const maxContainerRetries = 5
-	const healthCheckRetries = 10
-
-	pool, err := dockertest.NewPool("")
-	if err != nil {
-		t.Fatalf("Could not connect to docker. Reason: %v", err)
-	}
-
-	isInsecure := GetEnvOrDefault("EVENTSTORE_INSECURE", "true") == "true"
-	if !isInsecure {
-		err = setTLSContext(options)
-		if err != nil {
-			t.Fatal(err)
-		}
-	}
-
-	var resource *dockertest.Resource
-	var endpoint string
+	env := map[string]string{}
+	var files []testcontainers.ContainerFile
+	var insecure bool
```

Comment:
Not a big deal but it seems like we can simplify the logic below by using `GetEnvOrDefault` like elsewhere in the source file:

```go
...
	env := map[string]string{}
	var files []testcontainers.ContainerFile
	var insecure = GetEnvOrDefault("EVENTSTORE_INSECURE", "true") == "true"

	if insecure == false {
		err := verifyCertificatesExist()

		if err != nil {
			return nil, nil, err
		}

		certsDir, err := getCertificatesDir()

		if err != nil {
			return nil, nil, err
		}

		env["EVENTSTORE_CERTIFICATE_FILE"] = "/etc/eventstore/certs/node/node.crt"
		env["EVENTSTORE_CERTIFICATE_PRIVATE_KEY_FILE"] = "/etc/eventstore/certs/node/node.key"
		env["EVENTSTORE_TRUSTED_ROOT_CERTIFICATES_PATH"] = "/etc/eventstore/certs/ca"

		files = append(files, testcontainers.ContainerFile{
			HostFilePath:      certsDir,
			ContainerFilePath: "/etc/eventstore/certs",
			FileMode:          int64(0755),
		})
	}

	env["EVENTSTORE_INSECURE"] = strconv.FormatBool(insecure)
...
```

What do you think?